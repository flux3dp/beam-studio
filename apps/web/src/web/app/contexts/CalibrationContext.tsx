import React, { createContext, useState } from 'react';

import DeviceMaster from 'helpers/device-master';
import PreviewModeController from 'app/actions/beambox/preview-mode-controller';
import storage from 'implementations/storage';
import versionChecker from 'helpers/version-checker';
import { CameraConfig } from 'interfaces/Camera';
import {
  DEFAULT_CAMERA_OFFSET,
  STEP_ASK_READJUST,
  STEP_PUT_PAPER,
} from 'app/constants/camera-calibration-constants';
import { IDeviceInfo } from 'interfaces/IDevice';

interface CalibrationContextType {
  borderless: boolean;
  cameraPosition: { x: number; y: number };
  setCameraPosition: (val: { x: number; y: number }) => void;
  calibratedMachines: string[];
  setCalibratedMachines: (val: string[]) => void;
  currentOffset: CameraConfig;
  setCurrentOffset: (val: CameraConfig) => void;
  currentStep: symbol;
  setCurrentStep: (val: symbol) => void;
  lastConfig: CameraConfig;
  setLastConfig: (val: CameraConfig) => void;
  gotoNextStep: (step: symbol) => void;
  onClose: (completed: boolean) => void;
  originFanSpeed: number;
  setOriginFanSpeed: (val: number) => void;
  imgBlobUrl: string;
  setImgBlobUrl: (v: string) => void;
  device: IDeviceInfo;
  unit: string;
}

export const CalibrationContext = createContext<CalibrationContextType>({
  borderless: false,
  cameraPosition: { x: 0, y: 0 },
  setCameraPosition: () => {},
  calibratedMachines: [],
  setCalibratedMachines: () => {},
  currentOffset: null,
  setCurrentOffset: () => {},
  currentStep: STEP_PUT_PAPER,
  setCurrentStep: () => {},
  lastConfig: null,
  setLastConfig: () => {},
  gotoNextStep: () => {},
  onClose: () => {},
  imgBlobUrl: '',
  setImgBlobUrl: () => {},
  originFanSpeed: 100,
  setOriginFanSpeed: () => {},
  device: null,
  unit: 'mm',
});

interface CalibrationProviderProps {
  children: React.ReactNode;
  borderless: boolean;
  device: IDeviceInfo;
  onClose: (completed: boolean) => void;
}

let calibratedMachines = [];

export function CalibrationProvider({
  children,
  borderless,
  device,
  onClose,
}: CalibrationProviderProps): JSX.Element {
  const didCalibrate = calibratedMachines.includes(device.uuid);
  const initStep: symbol = didCalibrate ? STEP_ASK_READJUST : STEP_PUT_PAPER;
  const [currentStep, setCurrentStep] = useState(initStep);
  const [currentOffset, setCurrentOffset] = useState(DEFAULT_CAMERA_OFFSET);
  const [lastConfig, setLastConfig] = useState(DEFAULT_CAMERA_OFFSET);
  const setCalibratedMachines = (newList: string[]) => {
    calibratedMachines = newList;
  };
  const [imgBlobUrl, setImgBlobUrl] = useState('');
  const [cameraPosition, setCameraPosition] = useState({ x: 0, y: 0 });
  const [originFanSpeed, setOriginFanSpeed] = useState(1000);
  const unit = (storage.get('default-units') as string) || 'mm';

  const wrappedOnClose = async (completed: boolean) => {
    onClose(completed);
    await PreviewModeController.end({ shouldWaitForEnd: true });
    const tempCmdAvailable = versionChecker(device?.version).meetRequirement('TEMP_I2C_CMD');
    if (originFanSpeed && !tempCmdAvailable) {
      await DeviceMaster.setFan(originFanSpeed);
    }
  };

  return (
    <CalibrationContext.Provider
      value={{
        borderless,
        device,
        currentStep,
        setCurrentStep,
        cameraPosition,
        setCameraPosition,
        currentOffset,
        setCurrentOffset,
        lastConfig,
        setLastConfig,
        calibratedMachines,
        setCalibratedMachines,
        imgBlobUrl,
        setImgBlobUrl,
        originFanSpeed,
        setOriginFanSpeed,
        gotoNextStep: setCurrentStep,
        unit,
        onClose: wrappedOnClose,
      }}
    >
      {children}
    </CalibrationContext.Provider>
  );
}
