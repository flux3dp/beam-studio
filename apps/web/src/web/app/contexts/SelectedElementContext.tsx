import React, { createContext, memo, useCallback, useEffect, useState, useRef } from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';

const canvasEventEmitter = eventEmitterFactory.createEventEmitter('canvas');

export interface SelectedElementContextType {
  selectedElement: Element | null;
}

export const SelectedElementContext = createContext<SelectedElementContextType>({
  selectedElement: null,
});

interface Props {
  children: React.ReactNode;
}

export const SelectedElementContextProvider = memo(({ children }: Props): JSX.Element => {
  const [selectedElement, setSelectedElement] = useState<Element | null>(null);
  const selectedElementRef = useRef<Element | null>(null);

  const handleSetSelectedElem = useCallback(
    (elem: Element): void => {
      if (elem === selectedElementRef.current) return;
      selectedElementRef.current = elem;
      setSelectedElement((cur) => {
        if (cur === elem) return cur;
        (document.activeElement as HTMLInputElement).blur();
        return elem;
      });
    },
    []
  );

  useEffect(() => {
    canvasEventEmitter.on('SET_SELECTED_ELEMENT', handleSetSelectedElem);
    return () => {
      canvasEventEmitter.off('SET_SELECTED_ELEMENT', handleSetSelectedElem);
    };
  }, [handleSetSelectedElem]);

  return (
    <SelectedElementContext.Provider value={{ selectedElement }}>
      {children}
    </SelectedElementContext.Provider>
  );
});

export default {
  SelectedElementContext,
  SelectedElementContextProvider,
};
