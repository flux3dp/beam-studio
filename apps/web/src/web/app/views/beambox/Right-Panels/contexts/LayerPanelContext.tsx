import React, { createContext, useEffect, useState, useCallback, useMemo } from 'react';

import doLayersContainsVector from 'helpers/layer/check-vector';
import eventEmitterFactory from 'helpers/eventEmitterFactory';
import useForceUpdate from 'helpers/use-force-update';
import useWorkarea from 'helpers/hooks/useWorkarea';
import { getLayerElementByName } from 'helpers/layer/layer-helper';
import { promarkModels } from 'app/actions/beambox/constant';

interface ILayerPanelContext {
  selectedLayers: string[];
  setSelectedLayers: (selectedLayers: string[]) => void;
  forceUpdate: () => void;
  forceUpdateSelectedLayers: () => void;
  hasVector: boolean;
  hasGradient: boolean;
}

export const LayerPanelContext = createContext<ILayerPanelContext>({
  selectedLayers: [],
  setSelectedLayers: () => {},
  forceUpdate: () => {},
  forceUpdateSelectedLayers: () => {},
  hasVector: false,
  hasGradient: false,
});
const layerPanelEventEmitter = eventEmitterFactory.createEventEmitter('layer-panel');

interface Props {
  children?: React.ReactNode;
}

export const LayerPanelContextProvider = ({ children }: Props): JSX.Element => {
  const [hasVector, setHasVector] = useState<boolean>(false);
  const [hasGradient, setHasGradient] = useState<boolean>(false);
  const [selectedLayers, setSelectedLayers] = useState<string[]>([]);
  const forceUpdate = useForceUpdate();
  const forceUpdateSelectedLayers = useCallback(
    () => setSelectedLayers([...selectedLayers]),
    [selectedLayers]
  );
  const workarea = useWorkarea();
  const isPromark = useMemo(() => promarkModels.has(workarea), [workarea]);

  const lazySetSelectedLayers = useCallback(
    (newLayers: string[]) => {
      if (
        newLayers.length === selectedLayers.length &&
        newLayers.every((name, i) => name === selectedLayers[i])
      ) {
        return;
      }
      setSelectedLayers(newLayers);
    },
    [selectedLayers, setSelectedLayers]
  );

  useEffect(() => {
    layerPanelEventEmitter.on('UPDATE_LAYER_PANEL', forceUpdate);
    return () => {
      layerPanelEventEmitter.removeListener('UPDATE_LAYER_PANEL', forceUpdate);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    layerPanelEventEmitter.on('SET_SELECTED_LAYERS', lazySetSelectedLayers);
    return () => {
      layerPanelEventEmitter.removeListener('SET_SELECTED_LAYERS', lazySetSelectedLayers);
    };
  }, [lazySetSelectedLayers]);

  useEffect(() => {
    const getSelectedLayers = (response: { selectedLayers: string[] }) => {
      response.selectedLayers = selectedLayers;
    };
    layerPanelEventEmitter.on('GET_SELECTED_LAYERS', getSelectedLayers);
    const checkVector = () => {
      const newVal = doLayersContainsVector(selectedLayers);
      setHasVector(newVal);
    };
    layerPanelEventEmitter.on('CHECK_VECTOR', checkVector);
    return () => {
      layerPanelEventEmitter.removeListener('GET_SELECTED_LAYERS', getSelectedLayers);
      layerPanelEventEmitter.removeListener('CHECK_VECTOR', checkVector);
    };
  }, [selectedLayers]);

  useEffect(() => {
    const checkGradient = () => {
      if (isPromark) {
        const newVal = selectedLayers.some((layerName: string) => {
          const layer = getLayerElementByName(layerName);
          return !!layer?.querySelector('image[data-shading="true"]');
        });
        setHasGradient(newVal);
      }
    };
    checkGradient();
    layerPanelEventEmitter.on('CHECK_GRADIENT', checkGradient);
    return () => {
      layerPanelEventEmitter.removeListener('CHECK_GRADIENT', checkGradient);
    };
  }, [selectedLayers, isPromark]);

  // move doLayersContainsVector and setVector in effect to avoid heavy render process
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    const newVal = doLayersContainsVector(selectedLayers);
    if (hasVector !== newVal) {
      setHasVector(newVal);
    }
    if (isPromark) {
      const newGradientVal = selectedLayers.some((layerName: string) => {
        const layer = getLayerElementByName(layerName);
        return !!layer?.querySelector('image[data-shading="true"]');
      });
      if (newGradientVal !== hasGradient) setHasGradient(newGradientVal);
    }
  });

  return (
    <LayerPanelContext.Provider
      value={{
        selectedLayers,
        setSelectedLayers: lazySetSelectedLayers,
        forceUpdate,
        forceUpdateSelectedLayers,
        hasVector,
        hasGradient,
      }}
    >
      {children}
    </LayerPanelContext.Provider>
  );
};
