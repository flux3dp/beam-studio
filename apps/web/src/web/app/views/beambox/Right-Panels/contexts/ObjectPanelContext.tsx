import React, { useCallback, useEffect, useRef, useState } from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';
import useForceUpdate from 'helpers/use-force-update';

interface IObjectPanelContext {
  polygonSides: number;
  dimensionValues: any;
  activeKey: string | null;
  updateDimensionValues: (newValues: any) => void;
  getDimensionValues: (
    response: {
      dimensionValues: any;
    },
    key?: string
  ) => any;
  updateActiveKey: (activeKey: string | null) => void;
  updateObjectPanel: () => void;
}

export const ObjectPanelContext = React.createContext<IObjectPanelContext>({
  polygonSides: 5,
  dimensionValues: {},
  activeKey: null,
  updateDimensionValues: () => {},
  getDimensionValues: () => {},
  updateActiveKey: () => {},
  updateObjectPanel: () => {},
});
const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');
const minRenderInterval = 200;

interface Props {
  children?: React.ReactNode;
}

export const ObjectPanelContextProvider = ({ children }: Props): JSX.Element => {
  const forceUpdate = useForceUpdate();
  const [polygonSides, setPolygonSides] = useState(5);
  const [activeKey, setActiveKey] = useState<string | null>(null);
  const dimensionValues = useRef<{ [key: string]: number | string }>({});
  const lastUpdateTime = useRef(Date.now());
  const updateTimeout = useRef<NodeJS.Timeout>(null);
  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_POLYGON_SIDES', setPolygonSides);
    objectPanelEventEmitter.on('UPDATE_ACTIVE_KEY', setActiveKey);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_POLYGON_SIDES', setPolygonSides);
      objectPanelEventEmitter.removeListener('UPDATE_ACTIVE_KEY', setActiveKey);
    }
  }, []);

  const getDimensionValues = useCallback((response: { dimensionValues: any }, key?: string) => {
    response.dimensionValues = key ? dimensionValues.current[key] : dimensionValues.current;
  }, []);

  useEffect(() => {
    objectPanelEventEmitter.on('GET_DIMENSION_VALUES', getDimensionValues);
    return () => {
      objectPanelEventEmitter.removeListener('GET_DIMENSION_VALUES', getDimensionValues);
    }
  }, [getDimensionValues]);
  const updateDimensionValues = useCallback((newValues: { [key: string]: number | string }) => {
    dimensionValues.current = {
      ...dimensionValues.current,
      ...newValues,
    };
  }, []);

  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_DIMENSION_VALUES', updateDimensionValues);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_DIMENSION_VALUES', updateDimensionValues);
    }
  }, [updateDimensionValues]);
  const getActiveKey = useCallback((response: { activeKey: string | null }) => {
    response.activeKey = activeKey;
  }, [activeKey]);

  useEffect(() => {
    objectPanelEventEmitter.on('GET_ACTIVE_KEY', getActiveKey);
    return () => {
      objectPanelEventEmitter.removeListener('GET_ACTIVE_KEY', getActiveKey);
    }
  }, [getActiveKey]);

  const updateObjectPanel = useCallback(() => {
    clearTimeout(updateTimeout.current);
    const time = Date.now();
    if (time - lastUpdateTime.current >= minRenderInterval) {
      lastUpdateTime.current = time;
      forceUpdate();
    } else {
      updateTimeout.current = setTimeout(() => {
        lastUpdateTime.current += minRenderInterval;
        forceUpdate();
      }, lastUpdateTime.current + minRenderInterval - time);
    }
  }, [forceUpdate]);
  useEffect(() => {
    objectPanelEventEmitter.on('UPDATE_OBJECT_PANEL', updateObjectPanel);
    return () => {
      objectPanelEventEmitter.removeListener('UPDATE_OBJECT_PANEL', updateObjectPanel);
    }
  }, [updateObjectPanel]);

  return (
    <ObjectPanelContext.Provider
      value={{
        polygonSides,
        dimensionValues: dimensionValues.current,
        activeKey,
        updateDimensionValues,
        getDimensionValues,
        updateActiveKey: setActiveKey,
        updateObjectPanel,
      }}
    >
      {children}
    </ObjectPanelContext.Provider>
  );
};
