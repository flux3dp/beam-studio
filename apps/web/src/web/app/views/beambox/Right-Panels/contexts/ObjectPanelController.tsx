import eventEmitterFactory from 'helpers/eventEmitterFactory';

const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');

const updateDimensionValues = (newValue: any): void => {
  objectPanelEventEmitter.emit('UPDATE_DIMENSION_VALUES', newValue);
};

const getDimensionValues = (key?: string): any => {
  const response = {
    dimensionValues: {},
  };
  objectPanelEventEmitter.emit('GET_DIMENSION_VALUES', response, key);
  return response.dimensionValues;
};

const minEventInterval = 50;
let updateObjectPanelTimeout = null;
const updateObjectPanel = (): void => {
  clearTimeout(updateObjectPanelTimeout);
  updateObjectPanelTimeout = setTimeout(() => {
    objectPanelEventEmitter.emit('UPDATE_OBJECT_PANEL');
  }, minEventInterval);
};

const updatePolygonSides = (polygonSides: number): void => {
  objectPanelEventEmitter.emit('UPDATE_POLYGON_SIDES', polygonSides);
};

const updateActiveKey = (activeKey: string | null): void => {
  objectPanelEventEmitter.emit('UPDATE_ACTIVE_KEY', activeKey);
};

const getActiveKey = (): string | null => {
  const response = {
    activeKey: undefined,
  };
  objectPanelEventEmitter.emit('GET_ACTIVE_KEY', response);
  return response.activeKey;
};

export default {
  updateObjectPanel,
  updateDimensionValues,
  getDimensionValues,
  updatePolygonSides,
  updateActiveKey,
  getActiveKey,
  events: objectPanelEventEmitter,
};
