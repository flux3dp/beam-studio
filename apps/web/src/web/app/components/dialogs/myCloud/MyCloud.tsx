import React, { useContext, useMemo } from 'react';
import { Button, Modal, Space } from 'antd';

import browser from 'implementations/browser';
import FloatingPanel from 'app/widgets/FloatingPanel';
import FluxIcons from 'app/icons/flux/FluxIcons';
import isFluxPlusActive from 'helpers/is-flux-plus-active';
import layoutConstants from 'app/constants/layout-constants';
import useI18n from 'helpers/useI18n';
import { getCurrentUser } from 'helpers/api/flux-id';
import { MyCloudContext, MyCloudProvider } from 'app/contexts/MyCloudContext';
import { useIsMobile } from 'helpers/system-helper';

import GridFile from './GridFile';
import Head from './Head';
import styles from './MyCloud.module.scss';

interface Props {
  onClose: () => void;
}

const MyCloudModal = (): JSX.Element => {
  const user = getCurrentUser();
  const LANG = useI18n();
  const lang = LANG.my_cloud;
  const isMobile = useIsMobile();
  const { onClose, files, setSelectedId } = useContext(MyCloudContext);
  const anchors = [0, window.innerHeight - layoutConstants.menuberHeight];

  const content = useMemo(() => {
    if (files === undefined) return <div className={styles.placeholder}>{lang.loading_file}</div>;
    if (files.length === 0)
      return (
        <div className={styles.placeholder}>
          <div>{lang.no_file_title}</div>
          <div>{lang.no_file_subtitle}</div>
        </div>
      );
    return (
      <div className={styles.grids} onClick={() => setSelectedId(null)}>
        {files.map((file) => (
          <GridFile key={file.uuid} file={file} />
        ))}
      </div>
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [files]);

  const title = (
    <Space
      className={styles['title-container']}
      size="middle"
      align="center"
      direction={isMobile ? 'vertical' : 'horizontal'}
    >
      <div className={styles.title}>
        {user.info?.subscription?.is_valid && <FluxIcons.FluxPlus />}
        {lang.title}
      </div>
      {!user.info?.subscription?.is_valid && (
        <div className={styles.limit}>
          <div className={styles.tag}>
            {lang.file_limit} {files ? files.length : '?'}/5
          </div>
          {isFluxPlusActive && (
            <Button
              type="link"
              size="small"
              onClick={() => browser.open(LANG.flux_id_login.flux_plus.website_url)}
            >
              {lang.upgrade}
            </Button>
          )}
        </div>
      )}
    </Space>
  );

  return isMobile ? (
    <FloatingPanel
      className={styles.panel}
      anchors={anchors}
      title={title}
      fixedContent={<Head />}
      onClose={onClose}
    >
      {content}
    </FloatingPanel>
  ) : (
    <Modal title={title} footer={null} width={720} onCancel={onClose} centered open>
      <Head />
      {content}
    </Modal>
  );
};

const MyCloud = ({ onClose }: Props): JSX.Element => (
  <MyCloudProvider onClose={onClose}>
    <MyCloudModal />
  </MyCloudProvider>
);

export default MyCloud;
