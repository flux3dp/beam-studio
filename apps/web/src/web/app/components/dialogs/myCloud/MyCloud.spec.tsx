/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

import { IFile } from 'interfaces/IMyCloud';
import { MyCloudContext } from 'app/contexts/MyCloudContext';

import MyCloud from './MyCloud';

const mockFiles: IFile[] = [
  {
    uuid: 'mock-uuid-1111',
    name: 'File name',
    size: 5788,
    thumbnail_url: 'https://s3/url',
    workarea: 'fhexa1',
    created_at: '2024-01-09T04:14:36.801586Z',
    last_modified_at: '2024-01-09T06:42:04.824942Z',
  },
  {
    uuid: 'mock-uuid-2222',
    name: 'Another file',
    size: 5678,
    thumbnail_url: 'https://s3/url2',
    workarea: 'ado1',
    created_at: '2024-01-12T08:46:54.904853Z',
    last_modified_at: '2024-01-16T04:11:44.903500Z',
  },
];

jest.mock('helpers/useI18n', () => () => ({
  flux_id_login: {
    flux_plus: {
      website_url: 'https://website_url',
    },
  },
  my_cloud: {
    title: 'My Cloud',
    loading_file: 'Loading...',
    no_file_title: 'Save files to My Cloud to get started.',
    no_file_subtitle: 'Go to Menu > "File" > "Save to Cloud"',
    file_limit: 'Free file',
    upgrade: 'Upgrade',
  },
}));

const mockUser = {
  email: 'test123@gmail.com',
  info: {
    subscription: {
      is_valid: false,
    },
  },
};

const getCurrentUser = jest.fn();
jest.mock('helpers/api/flux-id', () => ({
  getCurrentUser: () => getCurrentUser(),
}));

const open = jest.fn();
jest.mock('implementations/browser', () => ({
  open: (...args) => open(...args),
}));

jest.mock('app/contexts/MyCloudContext', () => ({
  MyCloudContext: React.createContext({}),
  MyCloudProvider: ({ onClose, children }: any) => (
    <MyCloudContext.Provider value={{ onClose, files: mockFiles } as any}>
      {children}
    </MyCloudContext.Provider>
  ),
}));

const mockUseIsMobile = jest.fn();
jest.mock('helpers/system-helper', () => ({
  useIsMobile: () => mockUseIsMobile(),
}));

jest.mock('helpers/is-flux-plus-active', () => true);

jest.mock('./GridFile', () => ({ file }: any) => <div>Mock Grid File: {JSON.stringify(file)}</div>);
jest.mock('./Head', () => 'mock-head');

const mockOnClose = jest.fn();

describe('test MyCloud', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should rendered correctly', () => {
    getCurrentUser.mockReturnValue(mockUser);
    const { baseElement, getByText } = render(<MyCloud onClose={mockOnClose} />);
    expect(baseElement).toMatchSnapshot();
    fireEvent.click(getByText('Upgrade'));
    expect(open).toBeCalledTimes(1);
    expect(open).toBeCalledWith('https://website_url');
    const closeButton = baseElement.querySelector('.ant-modal-close');
    fireEvent.click(closeButton);
    expect(mockOnClose).toBeCalledTimes(1);
  });

  test('should rendered correctly when user is subscribed', () => {
    getCurrentUser.mockReturnValue({
      email: 'test123@gmail.com',
      info: {
        subscription: {
          is_valid: true,
        },
      },
    });
    const { baseElement } = render(<MyCloud onClose={mockOnClose} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('should rendered correctly in mobile', () => {
    getCurrentUser.mockReturnValue(mockUser);
    mockUseIsMobile.mockReturnValue(true);
    const { container } = render(<MyCloud onClose={mockOnClose} />);
    expect(container).toMatchSnapshot();
    const button = container.querySelector('.close-icon');
    fireEvent.click(button);
    waitFor(() => expect(mockOnClose).toBeCalledTimes(1));
  });
});
