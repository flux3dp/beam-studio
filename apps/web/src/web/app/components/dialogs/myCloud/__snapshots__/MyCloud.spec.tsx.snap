// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyCloud should rendered correctly 1`] = `
<body>
  <div
    class="adm-px-tester"
    style="--size: 10;"
  />
  <div
    class="adm-px-tester"
  />
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-hash"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-hash ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 720px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-hash ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle title-container"
                  >
                    <div
                      class="ant-space-item"
                    >
                      <div
                        class="title"
                      >
                        My Cloud
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div
                        class="limit"
                      >
                        <div
                          class="tag"
                        >
                          Free file
                           
                          2
                          /5
                        </div>
                        <button
                          class="ant-btn css-dev-only-do-not-override-hash ant-btn-link ant-btn-color-primary ant-btn-variant-link ant-btn-sm"
                          type="button"
                        >
                          <span>
                            Upgrade
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <mock-head />
                <div
                  class="grids"
                >
                  <div>
                    Mock Grid File: 
                    {"uuid":"mock-uuid-1111","name":"File name","size":5788,"thumbnail_url":"https://s3/url","workarea":"fhexa1","created_at":"2024-01-09T04:14:36.801586Z","last_modified_at":"2024-01-09T06:42:04.824942Z"}
                  </div>
                  <div>
                    Mock Grid File: 
                    {"uuid":"mock-uuid-2222","name":"Another file","size":5678,"thumbnail_url":"https://s3/url2","workarea":"ado1","created_at":"2024-01-12T08:46:54.904853Z","last_modified_at":"2024-01-16T04:11:44.903500Z"}
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`test MyCloud should rendered correctly in mobile 1`] = `
<div>
  <div
    class="adm-floating-panel panel panel"
    data-animating="true"
    style="height: 0px; transform: translateY(calc(100% + (0px)));"
  >
    <div
      class="adm-floating-panel-mask"
      style="display: none;"
    />
    <div
      class="adm-floating-panel-header"
    >
      <div
        class="adm-floating-panel-bar"
      />
    </div>
    <div
      class="adm-floating-panel-content"
    >
      <span
        class="anticon close-icon"
        role="img"
        tabindex="-1"
      >
        <svgr-url
          aria-hidden="true"
          classname=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        />
      </span>
      <div
        class="title"
      >
        <div
          class="ant-space css-dev-only-do-not-override-hash ant-space-vertical ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle title-container"
        >
          <div
            class="ant-space-item"
          >
            <div
              class="title"
            >
              My Cloud
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="limit"
            >
              <div
                class="tag"
              >
                Free file
                 
                2
                /5
              </div>
              <button
                class="ant-btn css-dev-only-do-not-override-hash ant-btn-link ant-btn-color-primary ant-btn-variant-link ant-btn-sm"
                type="button"
              >
                <span>
                  Upgrade
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <mock-head />
      <div
        class="scroll-content"
      >
        <div
          class="grids"
        >
          <div>
            Mock Grid File: 
            {"uuid":"mock-uuid-1111","name":"File name","size":5788,"thumbnail_url":"https://s3/url","workarea":"fhexa1","created_at":"2024-01-09T04:14:36.801586Z","last_modified_at":"2024-01-09T06:42:04.824942Z"}
          </div>
          <div>
            Mock Grid File: 
            {"uuid":"mock-uuid-2222","name":"Another file","size":5678,"thumbnail_url":"https://s3/url2","workarea":"ado1","created_at":"2024-01-12T08:46:54.904853Z","last_modified_at":"2024-01-16T04:11:44.903500Z"}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test MyCloud should rendered correctly when user is subscribed 1`] = `
<body>
  <div
    class="adm-px-tester"
    style="--size: 10;"
  />
  <div
    class="adm-px-tester"
  />
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-hash"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-hash ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 720px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-hash ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle title-container"
                  >
                    <div
                      class="ant-space-item"
                    >
                      <div
                        class="title"
                      >
                        <svgr-url />
                        My Cloud
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <mock-head />
                <div
                  class="grids"
                >
                  <div>
                    Mock Grid File: 
                    {"uuid":"mock-uuid-1111","name":"File name","size":5788,"thumbnail_url":"https://s3/url","workarea":"fhexa1","created_at":"2024-01-09T04:14:36.801586Z","last_modified_at":"2024-01-09T06:42:04.824942Z"}
                  </div>
                  <div>
                    Mock Grid File: 
                    {"uuid":"mock-uuid-2222","name":"Another file","size":5678,"thumbnail_url":"https://s3/url2","workarea":"ado1","created_at":"2024-01-12T08:46:54.904853Z","last_modified_at":"2024-01-16T04:11:44.903500Z"}
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
